app/code/Magento/Customer/Observer/AfterAddressSaveObserver.php

```diff
@@ -123,6 +123,7 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         /** @var $customerAddress Address */
         $customerAddress = $observer->getCustomerAddress();
         $customer = $customerAddress->getCustomer();
+        $customer->getAddressesCollection()->clear();
```

Comment:
@verma-ankur could you please explain how it is going to fix the problem?

Reply:
@orlangur When a group changed because of the valid or invalid vat id.
Then in customer old address attached. And the new changes in address ignores.
$customer->getAddressesCollection()->clear();
By doing this we are clearing the old address attached with the customer.
And the new changes will remain in address data.